### PR TITLE
Improve gitops/README.md to describe the process to setup a cluster using PaC

### PR DESCRIPTION
**Pull request by bnallapeta**

### CODE REVIEW

File: gitops/README.md [hunk 1]
```diff
@@ -41,18 +57,57 @@ b. Merge a PR with the kubeconfigs and overlay kustomization to trigger a Pipeli
...
+   - Do note that you need to install the GitHub app on your desired repository apart from the above two steps.
+   - PaC also supports the above steps on other Git providers such as Gitlab, Bitbucket all of which is documented [here](https://pipelinesascode.com/).
 
 Once the above two steps are done, the GitHub app (or any other Git provider app) you've set up from step 2 is now talking to PAC and would watch PRs and trigger PipelineRuns that automate the workflow previously described.
 
-The PR should have:
+In order to generate the kubeconfig for the cluster and kcp with the appropriate permissions using service accounts, we have created few scripts which are available [here](../images/access_setup/content/bin). The [README.md](../images/access_setup/README.md) describes the steps but essentially, you need the below two commands to generate kubeconfig for your cluster and kcp instance respectively.
+```
+- ./setup_compute.sh --kubeconfig /home/.kube/mycluster.kubeconfig --work_dir /repo/sre
+- ./setup_kcp.sh --kubeconfig /kcp/kcpinstance.kubeconfig --kcp_workspace plnsvc-ws --work_dir /repo/sre
```
Comment (Roming22) [inline gitops/README.md:84]: I'm not sure it's useful to suffix an object with it's type. Feel free to keep it that way if you disagree, as that might be a personal preference.

File: gitops/README.md [hunk 2]
```diff
@@ -41,18 +57,57 @@ b. Merge a PR with the kubeconfigs and overlay kustomization to trigger a Pipeli
...
+    - cluster-setup.yaml
+- credentials
+    - kubeconfig
+        - compute
+          - mycluster.kubeconfig    //kubeconfig of your cluster - generated by setup_compute.sh script
+        - kcp
+          - mykcp.kubeconfig        //kubeconfig of your kcp shared instance - generated by setup_kc[.sh script
+- environment
+    - compute
+      - mycluster                   //folder with the same name as your cluster
```
Comment (Roming22) [inline gitops/README.md:101]: The folder and its content are generated by setup_compute.sh as well. You can put a note saying that they can move the folder around and edit the default configuration in kustomization.yaml.

File: gitops/README.md [hunk 3]
```diff
@@ -41,18 +57,57 @@ b. Merge a PR with the kubeconfigs and overlay kustomization to trigger a Pipeli
...
+      This command will create a new Pipelines as Code Repository definition, a namespace where the pipelineruns run and configures webhook.
+   ```
+   - Do note that you need to install the GitHub app on your desired repository apart from the above two steps.
+   - PaC also supports the above steps on other Git providers such as Gitlab, Bitbucket all of which is documented [here](https://pipelinesascode.com/).
 
 Once the above two steps are done, the GitHub app (or any other Git provider app) you've set up from step 2 is now talking to PAC and would watch PRs and trigger PipelineRuns that automate the workflow previously described.
 
-The PR should have:
+In order to generate the kubeconfig for the cluster and kcp with the appropriate permissions using service accounts, we have created few scripts which are available [here](../images/access_setup/content/bin). The [README.md](../images/access_setup/README.md) describes the steps but essentially, you need the below two commands to generate kubeconfig for your cluster and kcp instance respectively.
+```
```
Comment (Roming22) [inline gitops/README.md:82]: This section contains steps that are also useful if running the setup manually.
I would that info earlier. Maybe we need subsections "### Prepare repository" and "### Deploy repository"?

**Comment by Roming22** [comment]
Let's make sure https://github.com/openshift-pipelines/pipelines-service/pull/109 is merged before this change goes in, as it references the new scripts.

File: gitops/README.md (hunk 1 above)
Comment (bnallapeta) [reply gitops/README.md:84]: Oh its just an example to make it obvious what to pass. I'm okay if you have anything else in mind. Do let me know.

File: gitops/README.md (hunk 2 above)
Comment (bnallapeta) [reply gitops/README.md:122]: Done.

File: gitops/README.md [hunk 4]
```diff
@@ -41,18 +57,57 @@ b. Merge a PR with the kubeconfigs and overlay kustomization to trigger a Pipeli
...
+   tkn pac create repo
+      This command will create a new Pipelines as Code Repository definition, a namespace where the pipelineruns run and configures webhook.
+   ```
+   - Do note that you need to install the GitHub app on your desired repository apart from the above two steps.
+   - PaC also supports the above steps on other Git providers such as Gitlab, Bitbucket all of which is documented [here](https://pipelinesascode.com/).
 
 Once the above two steps are done, the GitHub app (or any other Git provider app) you've set up from step 2 is now talking to PAC and would watch PRs and trigger PipelineRuns that automate the workflow previously described.
 
-The PR should have:
+In order to generate the kubeconfig for the cluster and kcp with the appropriate permissions using service accounts, we have created few scripts which are available [here](../images/access_setup/content/bin). The [README.md](../images/access_setup/README.md) describes the steps but essentially, you need the below two commands to generate kubeconfig for your cluster and kcp instance respectively.
```
[the remark this answers is not in the document]
Comment (bnallapeta) [reply gitops/README.md]: Done.

File: gitops/README.md [hunk 5]
```diff
@@ -23,36 +25,112 @@ Post this, a user will be able to log in to kcp workspace and start creating Pip
...
-b. Set up a Git application on any of the Git providers (GitHub, GitLab, Bitbucket etc.) by following the steps from [here](https://pipelinesascode.com).
+- Following the instructions from [here](https://pipelinesascode.com/docs/install/installation/) for a OpenShift cluster:
+  ```
+  kubectl patch tektonconfig config --type="merge" -p '{"spec": {"addon":{"enablePipelinesAsCode": false}}}'
+  kubectl apply -f https://raw.githubusercontent.com/openshift-pipelines/pipelines-as-code/stable/release.yaml
+  ```
+b. In order to set up PaC to listen to PRs, you must push the repository created in step 1 to a Git provider of your choice.
+
+c. From the path of your repo in your system, set up a Git application on any of the Git providers (GitHub, GitLab, Bitbucket etc.) by following the steps from [here](https://pipelinesascode.com).
+- As an example, if you wish to set up a GitHub app, we recommend using the [tkn pac cli tool](https://pipelinesascode.com/docs/guide/cli/) as it is easy to complete the whole process in just a couple of commands.
```
Comment (fgiloux) [inline gitops/README.md:94]: is it "we" or the PaC authors making this recommendation?
This is an imperative approach, which makes things easy but not declarative, hence not GitOps friendly

File: gitops/README.md [hunk 6]
```diff
@@ -23,36 +25,112 @@ Post this, a user will be able to log in to kcp workspace and start creating Pip
...
+- .tekton
+  - kcp-registration.yaml
+  - cluster-setup.yaml
+- credentials
+  - kubeconfig
+    - compute
+    - kcp
+- environment
+  - compute
+```
```
[the remark this answers is not in the document]
Comment (fgiloux) [reply gitops/README.md:51]: Also there is another approach of creating a repository that I have mentioned many times and is just to make a fork

**Comment by bnallapeta** [comment]
@fgiloux @Roming22  I have worked on all the changes requested by Frederic above. Now that #109 is merged, PTAL at this PR.

**Review by Roming22 — APPROVED** [review]

File: gitops/README.md [hunk 7]
```diff
@@ -23,36 +25,110 @@ Post this, a user will be able to log in to kcp workspace and start creating Pip
...
+- ./setup_compute.sh --kubeconfig /home/.kube/mycluster.kubeconfig --work_dir /repo/sre
+- ./setup_kcp.sh --kubeconfig /kcp/kcpinstance.kubeconfig --kcp_workspace plnsvc-ws --work_dir /repo/sre
+```
+
+### Run manual scripts
+
+Currently, we support two ways to get the cluster set up and ready to start creating PipelineRuns, one of which is to run the below scripts manually.  
+_Note: Please run these scripts only if you intend not to use PaC to automate the workflow._
+
```
[the remark this answers is not in the document]
Comment (bnallapeta) [reply gitops/README.md:69]: Yes, following this suggestion, I have placed this [here](https://github.com/openshift-pipelines/pipelines-service/blob/main/gitops/README.md#how-to-run). It now clearly shows Step 3 having Option 1 and Option 2.

**Pull request merged**